pkg/azuredisk/nodeserver.go

```diff
@@ -280,12 +280,10 @@ func (d *Driver) NodeGetCapabilities(ctx context.Context, req *csi.NodeGetCapabi
 func (d *Driver) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (*csi.NodeGetInfoResponse, error) {
 	klog.V(5).Infof("Using default NodeGetInfo")
 
-	instances, ok := d.cloud.Instances()
-	if !ok {
-		return nil, status.Error(codes.Internal, "Failed to get instances from cloud provider")
-	}
-
-	instanceType, err := instances.InstanceType(context.TODO(), types.NodeName(d.NodeID))
+	// Disable UseInstanceMetadata to mitigate a timeout issue from IMDS
+	// https://github.com/kubernetes-sigs/azuredisk-csi-driver/issues/168
+	d.cloud.UseInstanceMetadata = false
```

Comment:
why disable IMDS here?

Reply:
I received the following error on csi-azuredisk-controller when both csi-azuredisk-controller and csi-azuredisk-node have `hostNetwork: true` in their podspec:
```
Events:
  Type     Reason            Age                From               Message
  ----     ------            ----               ----               -------
  Warning  FailedScheduling  6s (x25 over 34s)  default-scheduler  0/3 nodes are available: 2 node(s) didn't have free ports for the requested pod ports
```
I suspected that only one pod per node can have `hostNetwork: true`. That's why I moved `hostNetwork: true` csi-azuredisk-node to csi-azuredisk-controller. With hostNetwork disabled, csi-azuredisk-node cannot reach IMDS due to the timeout issue described in #168. I manually disabled `UseInstanceMetadata` in the config so `d.cloud.InstanceType` can execute successfully.

Reply:
I think that's because node and controller ports conflict with each other, pls try with below patch, thanks.
```
diff --git a/deploy/csi-azuredisk-controller.yaml b/deploy/csi-azuredisk-controller.yaml
index e2c9ea1..56040c7 100644
--- a/deploy/csi-azuredisk-controller.yaml
+++ b/deploy/csi-azuredisk-controller.yaml
@@ -74,7 +74,7 @@ spec:
           args:
             - --csi-address=/csi/csi.sock
             - --connection-timeout=3s
-            - --health-port=9602
+            - --health-port=9601
           volumeMounts:
             - name: socket-dir
               mountPath: /csi
@@ -85,10 +85,10 @@ spec:
             - "--endpoint=$(CSI_ENDPOINT)"
             - "--nodeid=$(KUBE_NODE_NAME)"
           ports:
-            - containerPort: 9602
+            - containerPort: 9601
               name: healthz
               protocol: TCP
-            - containerPort: 10252
+            - containerPort: 19601
               name: metrics
               protocol: TCP
           livenessProbe:
@@ -133,6 +133,6 @@ spec:
   selector:
     app: csi-azuredisk-controller
   ports:
-  - port: 10252
-    targetPort: 10252
+  - port: 19601
+    targetPort: 19601
   type: ClusterIP
```